The job **`aarch64-gnu-llvm-19-2`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/15405843254/job/43348349399) [(plain)](https://github.com/rust-lang/rust/commit/00e75514f611318b497d589aaf2917ea6b6d9481/checks/43348349399/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
[RUSTC-TIMING] corebenches test:true 8.365
error: no rules expected `assert_eq_const_safe`
##[error]   --> coretests/tests/num/int_macros.rs:696:13
    |
1   | / macro_rules! int_module {
2   | |     ($T:ident, $U:ident) => {
3   | |         use core::num::ParseIntError;
4   | |         use core::ops::{BitAnd, BitOr, BitXor, Not, Shl, Shr};
...   |
696 | |             assert_eq_const_safe!(Option<$T>: <$T>::checked_exact_div(EXACT_DIV_SUCCESS_DIVIDEND1, EXACT_DIV_SUCCESS_DIVISOR1), Some(EXAC...
    | |             ^^^^^^^^^^^^^^^^^^^^ no rules expected this token in macro call
...   |
710 | |     };
711 | | }
    | |_- in this expansion of `int_module!`
    |
---

error: no rules expected `assert_eq_const_safe`
##[error]   --> coretests/tests/num/int_macros.rs:696:13
    |
1   | / macro_rules! int_module {
2   | |     ($T:ident, $U:ident) => {
3   | |         use core::num::ParseIntError;
4   | |         use core::ops::{BitAnd, BitOr, BitXor, Not, Shl, Shr};
...   |
696 | |             assert_eq_const_safe!(Option<$T>: <$T>::checked_exact_div(EXACT_DIV_SUCCESS_DIVIDEND1, EXACT_DIV_SUCCESS_DIVISOR1), Some(EXAC...
    | |             ^^^^^^^^^^^^^^^^^^^^ no rules expected this token in macro call
...   |
710 | |     };
711 | | }
    | |_- in this expansion of `int_module!`
    |
---

error: no rules expected `assert_eq_const_safe`
##[error]   --> coretests/tests/num/int_macros.rs:696:13
    |
1   | / macro_rules! int_module {
2   | |     ($T:ident, $U:ident) => {
3   | |         use core::num::ParseIntError;
4   | |         use core::ops::{BitAnd, BitOr, BitXor, Not, Shl, Shr};
...   |
696 | |             assert_eq_const_safe!(Option<$T>: <$T>::checked_exact_div(EXACT_DIV_SUCCESS_DIVIDEND1, EXACT_DIV_SUCCESS_DIVISOR1), Some(EXAC...
    | |             ^^^^^^^^^^^^^^^^^^^^ no rules expected this token in macro call
...   |
710 | |     };
711 | | }
    | |_- in this expansion of `int_module!`
    |
---

error: no rules expected `assert_eq_const_safe`
##[error]   --> coretests/tests/num/int_macros.rs:696:13
    |
1   | / macro_rules! int_module {
2   | |     ($T:ident, $U:ident) => {
3   | |         use core::num::ParseIntError;
4   | |         use core::ops::{BitAnd, BitOr, BitXor, Not, Shl, Shr};
...   |
696 | |             assert_eq_const_safe!(Option<$T>: <$T>::checked_exact_div(EXACT_DIV_SUCCESS_DIVIDEND1, EXACT_DIV_SUCCESS_DIVISOR1), Some(EXAC...
    | |             ^^^^^^^^^^^^^^^^^^^^ no rules expected this token in macro call
...   |
710 | |     };
711 | | }
    | |_- in this expansion of `int_module!`
    |
---

error: no rules expected `assert_eq_const_safe`
##[error]   --> coretests/tests/num/int_macros.rs:696:13
    |
1   | / macro_rules! int_module {
2   | |     ($T:ident, $U:ident) => {
3   | |         use core::num::ParseIntError;
4   | |         use core::ops::{BitAnd, BitOr, BitXor, Not, Shl, Shr};
...   |
696 | |             assert_eq_const_safe!(Option<$T>: <$T>::checked_exact_div(EXACT_DIV_SUCCESS_DIVIDEND1, EXACT_DIV_SUCCESS_DIVISOR1), Some(EXAC...
    | |             ^^^^^^^^^^^^^^^^^^^^ no rules expected this token in macro call
...   |
710 | |     };
711 | | }
    | |_- in this expansion of `int_module!`
    |
---

error: no rules expected `assert_eq_const_safe`
##[error]   --> coretests/tests/num/uint_macros.rs:529:13
    |
1   | / macro_rules! uint_module {
2   | |     ($T:ident) => {
3   | |         use core::num::ParseIntError;
4   | |         use core::ops::{BitAnd, BitOr, BitXor, Not, Shl, Shr};
...   |
529 | |             assert_eq_const_safe!(Option<$T>: <$T>::checked_exact_div(EXACT_DIV_SUCCESS_DIVIDEND1, EXACT_DIV_SUCCESS_DIVISOR1), Some(EXAC...
    | |             ^^^^^^^^^^^^^^^^^^^^ no rules expected this token in macro call
...   |
542 | |     };
543 | | }
    | |_- in this expansion of `uint_module!`
    |
   ::: coretests/tests/lib.rs:123:1
    |
123 |   macro_rules! test_runtime_and_compiletime {
    |   ----------------------------------------- when calling this macro
    |
   ::: coretests/tests/num/u128.rs:1:1
    |
1   |   uint_module!(u128);
    |   ------------------ in this macro invocation
    |
    = note: while trying to match sequence start

error: no rules expected `assert_eq_const_safe`
##[error]   --> coretests/tests/num/uint_macros.rs:529:13
    |
1   | / macro_rules! uint_module {
2   | |     ($T:ident) => {
3   | |         use core::num::ParseIntError;
4   | |         use core::ops::{BitAnd, BitOr, BitXor, Not, Shl, Shr};
...   |
529 | |             assert_eq_const_safe!(Option<$T>: <$T>::checked_exact_div(EXACT_DIV_SUCCESS_DIVIDEND1, EXACT_DIV_SUCCESS_DIVISOR1), Some(EXAC...
    | |             ^^^^^^^^^^^^^^^^^^^^ no rules expected this token in macro call
...   |
542 | |     };
543 | | }
    | |_- in this expansion of `uint_module!`
    |
   ::: coretests/tests/lib.rs:123:1
    |
123 |   macro_rules! test_runtime_and_compiletime {
    |   ----------------------------------------- when calling this macro
    |
   ::: coretests/tests/num/u16.rs:1:1
    |
1   |   uint_module!(u16);
    |   ----------------- in this macro invocation
    |
    = note: while trying to match sequence start

error: no rules expected `assert_eq_const_safe`
##[error]   --> coretests/tests/num/uint_macros.rs:529:13
    |
1   | / macro_rules! uint_module {
2   | |     ($T:ident) => {
3   | |         use core::num::ParseIntError;
4   | |         use core::ops::{BitAnd, BitOr, BitXor, Not, Shl, Shr};
...   |
529 | |             assert_eq_const_safe!(Option<$T>: <$T>::checked_exact_div(EXACT_DIV_SUCCESS_DIVIDEND1, EXACT_DIV_SUCCESS_DIVISOR1), Some(EXAC...
    | |             ^^^^^^^^^^^^^^^^^^^^ no rules expected this token in macro call
...   |
542 | |     };
543 | | }
    | |_- in this expansion of `uint_module!`
    |
   ::: coretests/tests/lib.rs:123:1
    |
123 |   macro_rules! test_runtime_and_compiletime {
    |   ----------------------------------------- when calling this macro
    |
   ::: coretests/tests/num/u32.rs:1:1
    |
1   |   uint_module!(u32);
    |   ----------------- in this macro invocation
    |
    = note: while trying to match sequence start

error: no rules expected `assert_eq_const_safe`
##[error]   --> coretests/tests/num/uint_macros.rs:529:13
    |
1   | / macro_rules! uint_module {
2   | |     ($T:ident) => {
3   | |         use core::num::ParseIntError;
4   | |         use core::ops::{BitAnd, BitOr, BitXor, Not, Shl, Shr};
...   |
529 | |             assert_eq_const_safe!(Option<$T>: <$T>::checked_exact_div(EXACT_DIV_SUCCESS_DIVIDEND1, EXACT_DIV_SUCCESS_DIVISOR1), Some(EXAC...
    | |             ^^^^^^^^^^^^^^^^^^^^ no rules expected this token in macro call
...   |
542 | |     };
543 | | }
    | |_- in this expansion of `uint_module!`
    |
   ::: coretests/tests/lib.rs:123:1
    |
123 |   macro_rules! test_runtime_and_compiletime {
    |   ----------------------------------------- when calling this macro
    |
   ::: coretests/tests/num/u64.rs:1:1
    |
1   |   uint_module!(u64);
    |   ----------------- in this macro invocation
    |
    = note: while trying to match sequence start

error: no rules expected `assert_eq_const_safe`
##[error]   --> coretests/tests/num/uint_macros.rs:529:13
    |
1   | / macro_rules! uint_module {
2   | |     ($T:ident) => {
3   | |         use core::num::ParseIntError;
4   | |         use core::ops::{BitAnd, BitOr, BitXor, Not, Shl, Shr};
...   |
529 | |             assert_eq_const_safe!(Option<$T>: <$T>::checked_exact_div(EXACT_DIV_SUCCESS_DIVIDEND1, EXACT_DIV_SUCCESS_DIVISOR1), Some(EXAC...
    | |             ^^^^^^^^^^^^^^^^^^^^ no rules expected this token in macro call
...   |
542 | |     };
543 | | }
    | |_- in this expansion of `uint_module!`
    |
   ::: coretests/tests/lib.rs:123:1
    |
123 |   macro_rules! test_runtime_and_compiletime {
    |   ----------------------------------------- when calling this macro
    |
   ::: coretests/tests/num/u8.rs:1:1
    |
1   |   uint_module!(u8);
    |   ---------------- in this macro invocation
    |
    = note: while trying to match sequence start

[RUSTC-TIMING] alloctests test:true 13.350
[RUSTC-TIMING] alloctests test:true 14.220
[RUSTC-TIMING] coretests test:true 44.354
error: could not compile `coretests` (test "coretests") due to 10 previous errors
env -u RUSTC_WRAPPER CARGO_ENCODED_RUSTDOCFLAGS="-Csymbol-mangling-version=v0\u{1f}-Zrandomize-layout\u{1f}-Zunstable-options\u{1f}--check-cfg=cfg(bootstrap)\u{1f}--check-cfg=cfg(llvm_enzyme)\u{1f}-Dwarnings\u{1f}-Wrustdoc::invalid_codeblock_attributes\u{1f}--crate-version\u{1f}1.89.0-nightly\t(dce27221b\t2025-06-02)" CARGO_ENCODED_RUSTFLAGS="-Csymbol-mangling-version=v0\u{1f}-Zrandomize-layout\u{1f}-Zunstable-options\u{1f}--check-cfg=cfg(bootstrap)\u{1f}--check-cfg=cfg(llvm_enzyme)\u{1f}-Zmacro-backtrace\u{1f}-Csplit-debuginfo=off\u{1f}-Clink-arg=-L/usr/lib/llvm-19/lib\u{1f}-Cllvm-args=-import-instr-limit=10\u{1f}-Clink-args=-Wl,-z,origin\u{1f}-Clink-args=-Wl,-rpath,$ORIGIN/../lib\u{1f}-Alinker-messages\u{1f}--cap-lints=allow\u{1f}--cfg\u{1f}randomized_layouts" RUSTC="/checkout/obj/build/aarch64-unknown-linux-gnu/stage1-tools/cg_clif/dist/rustc-clif" RUSTDOC="/checkout/obj/build/aarch64-unknown-linux-gnu/stage1-tools/cg_clif/dist/rustdoc-clif" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage0/bin/cargo" "test" "--manifest-path" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage1-tools/cg_clif/build/sysroot_tests/Cargo.toml" "--target-dir" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage1-tools/cg_clif/build/sysroot_tests_target" "--locked" "--target" "aarch64-unknown-linux-gnu" "-p" "coretests" "-p" "alloctests" "--tests" "--" "-q" exited with status ExitStatus(unix_wait_status(25856))
Command has failed. Rerun with -v to see more details.
Build completed unsuccessfully in 0:10:40
  local time: Tue Jun  3 01:01:16 UTC 2025
  network time: Tue, 03 Jun 2025 01:01:16 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/coretests/tests/lib.rs
+++ b/library/coretests/tests/lib.rs
@@ -30,6 +30,7 @@
 #![feature(duration_constructors)]
 #![feature(duration_constructors_lite)]
 #![feature(error_generic_member_access)]
+#![feature(exact_div)]
 #![feature(exact_size_is_empty)]
 #![feature(extend_one)]
 #![feature(extern_types)]

--- a/library/coretests/tests/num/int_macros.rs
+++ b/library/coretests/tests/num/int_macros.rs
@@ -690,6 +690,12 @@ macro_rules! int_module {
         const EXACT_DIV_SUCCESS_DIVIDEND2: $T = 18;
         const EXACT_DIV_SUCCESS_DIVISOR2: $T = 3;
         const EXACT_DIV_SUCCESS_QUOTIENT2: $T = 6;
+        const EXACT_DIV_SUCCESS_DIVIDEND3: $T = -91;
+        const EXACT_DIV_SUCCESS_DIVISOR3: $T = 13;
+        const EXACT_DIV_SUCCESS_QUOTIENT3: $T = -7;
+        const EXACT_DIV_SUCCESS_DIVIDEND4: $T = -57;
+        const EXACT_DIV_SUCCESS_DIVISOR4: $T = -3;
+        const EXACT_DIV_SUCCESS_QUOTIENT4: $T = 19;
 
         test_runtime_and_compiletime! {
             fn test_exact_div() {
@@ -700,6 +706,14 @@ macro_rules! int_module {
                 // 18 / 3
                 assert_eq_const_safe!(Option<$T>: <$T>::checked_exact_div(EXACT_DIV_SUCCESS_DIVIDEND2, EXACT_DIV_SUCCESS_DIVISOR2), Some(EXACT_DIV_SUCCESS_QUOTIENT2));
                 assert_eq_const_safe!($T: <$T>::exact_div(EXACT_DIV_SUCCESS_DIVIDEND2, EXACT_DIV_SUCCESS_DIVISOR2), EXACT_DIV_SUCCESS_QUOTIENT2);
+
+                // -91 / 13
+                assert_eq_const_safe!(Option<$T>: <$T>::checked_exact_div(EXACT_DIV_SUCCESS_DIVIDEND3, EXACT_DIV_SUCCESS_DIVISOR3), Some(EXACT_DIV_SUCCESS_QUOTIENT3));
+                assert_eq_const_safe!($T: <$T>::exact_div(EXACT_DIV_SUCCESS_DIVIDEND3, EXACT_DIV_SUCCESS_DIVISOR3), EXACT_DIV_SUCCESS_QUOTIENT3);
+
+                // -57 / -3
+                assert_eq_const_safe!(Option<$T>: <$T>::checked_exact_div(EXACT_DIV_SUCCESS_DIVIDEND4, EXACT_DIV_SUCCESS_DIVISOR4), Some(EXACT_DIV_SUCCESS_QUOTIENT4));
+                assert_eq_const_safe!($T: <$T>::exact_div(EXACT_DIV_SUCCESS_DIVIDEND4, EXACT_DIV_SUCCESS_DIVISOR4), EXACT_DIV_SUCCESS_QUOTIENT4);
 
                 // failures
                 assert_eq_const_safe!(Option<$T>: <$T>::checked_exact_div(1, 2), None);

--- a/library/coretests/tests/num/int_macros.rs
+++ b/library/coretests/tests/num/int_macros.rs
@@ -683,5 +683,29 @@ macro_rules! int_module {
                 assert_eq_const_safe!($T: <$T>::unbounded_shr(17, SHIFT_AMOUNT_OVERFLOW3), 0);
             }
         }
+
+        const EXACT_DIV_SUCCESS_DIVIDEND1: $T = 42;
+        const EXACT_DIV_SUCCESS_DIVISOR1: $T = 6;
+        const EXACT_DIV_SUCCESS_QUOTIENT1: $T = 7;
+        const EXACT_DIV_SUCCESS_DIVIDEND2: $T = 18;
+        const EXACT_DIV_SUCCESS_DIVISOR2: $T = 3;
+        const EXACT_DIV_SUCCESS_QUOTIENT2: $T = 6;
+
+        test_runtime_and_compiletime! {
+            fn test_exact_div() {
+                // 42 / 6
+                assert_eq_const_safe!(Option<$T>: <$T>::checked_exact_div(EXACT_DIV_SUCCESS_DIVIDEND1, EXACT_DIV_SUCCESS_DIVISOR1), Some(EXACT_DIV_SUCCESS_QUOTIENT1));
+                assert_eq_const_safe!($T: <$T>::exact_div(EXACT_DIV_SUCCESS_DIVIDEND1, EXACT_DIV_SUCCESS_DIVISOR1), EXACT_DIV_SUCCESS_QUOTIENT1);
+
+                // 18 / 3
+                assert_eq_const_safe!(Option<$T>: <$T>::checked_exact_div(EXACT_DIV_SUCCESS_DIVIDEND2, EXACT_DIV_SUCCESS_DIVISOR2), Some(EXACT_DIV_SUCCESS_QUOTIENT2));
+                assert_eq_const_safe!($T: <$T>::exact_div(EXACT_DIV_SUCCESS_DIVIDEND2, EXACT_DIV_SUCCESS_DIVISOR2), EXACT_DIV_SUCCESS_QUOTIENT2);
+
+                // failures
+                assert_eq_const_safe!(Option<$T>: <$T>::checked_exact_div(1, 2), None);
+                assert_eq_const_safe!(Option<$T>: <$T>::checked_exact_div(<$T>::MIN, -1), None);
+                assert_eq_const_safe!(Option<$T>: <$T>::checked_exact_div(0, 0), None);
+            }
+        }
     };
 }

--- a/library/coretests/tests/num/uint_macros.rs
+++ b/library/coretests/tests/num/uint_macros.rs
@@ -516,5 +516,28 @@ macro_rules! uint_module {
                 assert_eq_const_safe!($T: <$T>::unbounded_shr(17, SHIFT_AMOUNT_OVERFLOW3), 0);
             }
         }
+
+        const EXACT_DIV_SUCCESS_DIVIDEND1: $T = 42;
+        const EXACT_DIV_SUCCESS_DIVISOR1: $T = 6;
+        const EXACT_DIV_SUCCESS_QUOTIENT1: $T = 7;
+        const EXACT_DIV_SUCCESS_DIVIDEND2: $T = 18;
+        const EXACT_DIV_SUCCESS_DIVISOR2: $T = 3;
+        const EXACT_DIV_SUCCESS_QUOTIENT2: $T = 6;
+
+        test_runtime_and_compiletime! {
+            fn test_exact_div() {
+                // 42 / 6
+                assert_eq_const_safe!(Option<$T>: <$T>::checked_exact_div(EXACT_DIV_SUCCESS_DIVIDEND1, EXACT_DIV_SUCCESS_DIVISOR1), Some(EXACT_DIV_SUCCESS_QUOTIENT1));
+                assert_eq_const_safe!($T: <$T>::exact_div(EXACT_DIV_SUCCESS_DIVIDEND1, EXACT_DIV_SUCCESS_DIVISOR1), EXACT_DIV_SUCCESS_QUOTIENT1);
+
+                // 18 / 3
+                assert_eq_const_safe!(Option<$T>: <$T>::checked_exact_div(EXACT_DIV_SUCCESS_DIVIDEND2, EXACT_DIV_SUCCESS_DIVISOR2), Some(EXACT_DIV_SUCCESS_QUOTIENT2));
+                assert_eq_const_safe!($T: <$T>::exact_div(EXACT_DIV_SUCCESS_DIVIDEND2, EXACT_DIV_SUCCESS_DIVISOR2), EXACT_DIV_SUCCESS_QUOTIENT2);
+
+                // failures
+                assert_eq_const_safe!(Option<$T>: <$T>::checked_exact_div(1, 2), None);
+                assert_eq_const_safe!(Option<$T>: <$T>::checked_exact_div(0, 0), None);
+            }
+        }
     };
 }

